sdk/cosmos/azure-cosmos/pom.xml

```diff
@@ -102,6 +102,19 @@ Licensed under the MIT License.
       </exclusions>
     </dependency>
 
+    <!-- Added this provided dependency to include necessary annotations used by "reactor-core".
+         Without this dependency, javadoc throws a warning as it cannot find enum When.MAYBE
+         which is used in @Nullable annotation in reactor core classes.
+
+         Similar provided dependency exits for "azure-core" as well.
+    -->
+    <dependency>
+      <groupId>com.google.code.findbugs</groupId>
+      <artifactId>jsr305</artifactId>
+      <version>3.0.2</version> <!-- {x-version-update;com.google.code.findbugs:jsr305;external_dependency} -->
+      <scope>provided</scope>
+    </dependency>
```

Comment:
azure-core reference: https://github.com/Azure/azure-sdk-for-java/blob/dea52be2efa93c5abecfdd0582b03a9e046cc957/sdk/core/azure-core/pom.xml#L73